gms/gossiper.cc

```diff
@@ -1047,7 +1047,7 @@ void gossiper::run() {
             }
 
             {
-                auto lock = get_units(_endpoint_update_semaphore, 1).get();
+                auto lock = lock_endpoint_update_semaphore().get();
```

Comment:
Do we still need to call replicate_live_endpoints_on_change in the gossip loop here? I think we do not need it anymore because every time _live_endpoints is accessed,  we replicate them to all shards already.

Reply:
You're right. This isn't needed after using `mutate_live_and_unreachable_endpoints`.
I'll add a patch to remove it.